backend/cms/models/page.py

```diff
@@ -142,13 +166,38 @@ class PageTranslation(models.Model):
     created_date = models.DateTimeField(default=timezone.now)
     last_updated = models.DateTimeField(auto_now=True)
 
+    @property
+    def ancestor_path(self):
+        return '/'.join([
+            ancestor.page_translations.get(language=self.language).slug
+            for ancestor in self.page.get_ancestors()
+        ])
+
     @property
     def permalink(self):
-        permalink = self.page.site.slug + '/'
-        permalink += self.language.code + '/'
-        for ancestor in self.page.get_ancestors(include_self=True):
-            permalink += ancestor.page_translations.get(language=self.language).slug + '/'
-        return permalink
+        return '{}/{}/{}/{}'.format(
```

Comment:
We can define the permalink without the ancestors in the path. That would make things considerably easier, if the structure changes.

Reply:
Hmm. I think the advantage of having informative URLs which tell me in which category the page lives outweighs the effort of managing multi-level permalinks. The hard part is to create a consistent database of permalinks, it shouldn't make much of a difference if we have to additionally store the ancestor path there. So I'd prefer having the ancestor path in the permalink...